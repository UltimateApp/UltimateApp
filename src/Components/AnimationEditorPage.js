import React from 'react';
import { StyleSheet, View, Text } from 'react-native';

import AnimationEditor from './animation/AnimationEditor';
import drillSquare from './animation/DrillSquare';

import theme from '../styles/theme.style';

export const AnimationEditorPage = props => {
  return (
    <View style={styles.animationEditorPage}>
      <AnimationEditor animation={drillSquare} />
    </View>
  );
};

export default AnimationEditorPage;

const styles = StyleSheet.create({
  animationEditorPage: {
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    flex: 1,
    width: '100%',
  },
});
