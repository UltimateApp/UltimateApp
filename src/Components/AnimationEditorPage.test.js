import React from 'react';
import { create, act } from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../Store/testStore';

import AnimationEditorPage from './AnimationEditorPage';

jest.mock('react-native-reanimated', () => require('react-native-reanimated/mock'));

describe('<AnimationEditorPage />', () => {
  it('renders correctly', async () => {
    const navigation = { navigate: jest.fn(), setOptions: jest.fn() };

    const tree = create(
      <Provider store={store}>
        <AnimationEditorPage navigation={navigation} />
      </Provider>,
    ).toJSON();
    await act(async () => expect(tree).toMatchSnapshot());
  });
});
