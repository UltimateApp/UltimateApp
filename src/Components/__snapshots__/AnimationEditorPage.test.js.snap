// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AnimationEditorPage /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "alignItems": "stretch",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "white",
            "flexDirection": "row",
            "justifyContent": "center",
          },
          Object {
            "height": 952.8571428571429,
          },
          Object {
            "width": 750,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "bottom": 0,
              "position": "absolute",
              "top": 0,
            },
            Object {
              "height": 952.8571428571429,
              "width": 750,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "height": 809.9285714285714,
              },
              Object {
                "width": 3.375,
              },
              Object {
                "borderRadius": 1,
              },
              Object {
                "borderWidth": 1,
              },
              Object {
                "backgroundColor": "black",
              },
              Object {
                "position": "absolute",
              },
              Object {
                "top": 47.642857142857146,
              },
              Object {
                "left": 712.5,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "height": 809.9285714285714,
              },
              Object {
                "width": 3.375,
              },
              Object {
                "borderRadius": 1,
              },
              Object {
                "borderWidth": 1,
              },
              Object {
                "backgroundColor": "gray",
              },
              Object {
                "position": "absolute",
              },
              Object {
                "top": 47.642857142857146,
              },
              Object {
                "left": 37.5,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "height": 3.375,
              },
              Object {
                "width": 675,
              },
              Object {
                "borderRadius": 1,
              },
              Object {
                "borderWidth": 1,
              },
              Object {
                "backgroundColor": "gray",
              },
              Object {
                "position": "absolute",
              },
              Object {
                "top": 47.642857142857146,
              },
              Object {
                "left": 37.5,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "height": 3.375,
              },
              Object {
                "width": 675,
              },
              Object {
                "borderRadius": 1,
              },
              Object {
                "borderWidth": 1,
              },
              Object {
                "backgroundColor": "gray",
              },
              Object {
                "position": "absolute",
              },
              Object {
                "top": 371.61428571428576,
              },
              Object {
                "left": 37.5,
              },
            ]
          }
        />
        <View
          height={33.75}
          style={
            Array [
              Object {
                "position": "absolute",
              },
              Object {
                "left": 358.125,
              },
              Object {
                "top": 605.1407142857144,
              },
              Object {
                "borderRadius": 16.875,
              },
              Object {
                "borderWidth": 1,
              },
              Object {
                "borderColor": "gray",
              },
              Object {
                "backgroundColor": "white",
              },
            ]
          }
          width={33.75}
        />
      </View>
      <View
        height="100%"
        style={
          Array [
            Object {
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
          ]
        }
        width="100%"
      />
      <View
        height="100%"
        style={
          Array [
            Object {
              "left": 0,
              "position": "absolute",
              "top": 0,
            },
          ]
        }
        width="100%"
      >
        <View
          style={
            Array [
              Object {
                "backgroundColor": "gray",
              },
              Object {
                "borderRadius": 10,
                "borderWidth": 0,
                "bottom": 7.5,
                "color": "white",
                "height": 15,
                "includeFontPadding": false,
                "position": "absolute",
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 15,
              },
              Object {
                "left": 10,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "backgroundColor": "gray",
              },
              Object {
                "borderRadius": 10,
                "borderWidth": 0,
                "bottom": 7.5,
                "color": "white",
                "height": 15,
                "includeFontPadding": false,
                "position": "absolute",
                "textAlign": "center",
                "textAlignVertical": "center",
                "width": 15,
              },
              Object {
                "left": 317.5,
              },
            ]
          }
        />
        <Text
          style={
            Array [
              Object {
                "bottom": 23.5,
                "position": "absolute",
              },
              Object {
                "left": 14,
              },
            ]
          }
        >
          1
        </Text>
        <Text
          style={
            Array [
              Object {
                "bottom": 23.5,
                "position": "absolute",
              },
              Object {
                "left": 321.5,
              },
            ]
          }
        >
          2
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "gray",
              },
              Object {
                "bottom": 13,
                "height": 4,
                "left": 20,
                "position": "absolute",
              },
              false,
              Object {
                "width": 615,
              },
            ]
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "black",
              "borderRadius": 10,
              "borderWidth": 0,
              "bottom": 7.5,
              "color": "white",
              "height": 15,
              "includeFontPadding": false,
              "left": 10,
              "opacity": 1,
              "position": "absolute",
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 15,
            }
          }
        />
        <View
          style={
            Object {
              "backgroundColor": "black",
              "borderRadius": 10,
              "borderWidth": 0,
              "bottom": 7.5,
              "color": "white",
              "height": 15,
              "includeFontPadding": false,
              "left": 317.5,
              "opacity": 0,
              "position": "absolute",
              "textAlign": "center",
              "textAlignVertical": "center",
              "width": 15,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "borderRadius": 10,
              "borderWidth": 0,
              "bottom": 0,
              "height": 30,
              "justifyContent": "center",
              "left": 625,
              "opacity": 1,
              "position": "absolute",
              "width": 30,
            }
          }
        >
          <Text />
        </View>
        <View
          style={
            Object {
              "backgroundColor": "black",
              "bottom": 13,
              "height": 4,
              "left": 20,
              "position": "absolute",
              "width": 0,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "bottom": 0,
              "height": 25,
              "left": 2.5,
              "opacity": 1,
              "position": "absolute",
              "width": 25,
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "bottom": 0,
              "height": 25,
              "left": 310,
              "opacity": 1,
              "position": "absolute",
              "width": 25,
            }
          }
        />
        <View
          height="100%"
          style={
            Array [
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          width="100%"
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 10,
                "borderWidth": 0,
                "bottom": 5.5,
                "opacity": 1,
                "position": "absolute",
                "right": 40,
              }
            }
          >
            <Text />
          </View>
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "borderRadius": 10,
                "borderWidth": 0,
                "bottom": 5.5,
                "opacity": 1,
                "position": "absolute",
                "right": 10,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 50,
          "marginHorizontal": 30,
          "marginTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "flex-start",
          }
        }
      >
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#2a9e91",
              "borderRadius": 0,
              "height": 0,
              "justifyContent": "center",
              "marginRight": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
              "width": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "bold",
              }
            }
          >
            1
          </Text>
        </View>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#AFAFAF",
              "borderRadius": 0,
              "height": 0,
              "justifyContent": "center",
              "marginRight": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
              "width": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontWeight": "bold",
              }
            }
          >
            1
          </Text>
        </View>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderColor": "black",
              "borderRadius": 0,
              "borderWidth": 0,
              "height": 0,
              "justifyContent": "center",
              "marginRight": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
              "width": 0,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "black",
                "fontWeight": "bold",
              }
            }
          >
            1
          </Text>
        </View>
        <View
          onMoveShouldSetResponder={[Function]}
          onMoveShouldSetResponderCapture={[Function]}
          onResponderEnd={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderReject={[Function]}
          onResponderRelease={[Function]}
          onResponderStart={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onStartShouldSetResponderCapture={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomColor": "#FEA500",
              "borderBottomWidth": 0,
              "borderColor": "transparent",
              "borderLeftWidth": 0,
              "borderRightWidth": 0,
              "borderStyle": "solid",
              "justifyContent": "center",
              "marginRight": 10,
              "textAlign": "center",
              "textAlignVertical": "center",
              "transform": Array [
                Object {
                  "translateX": 0,
                },
                Object {
                  "translateY": 0,
                },
              ],
            }
          }
        >
          <Text
            style={
              Object {
                "height": 0,
                "width": 0,
              }
            }
          >
            1
          </Text>
        </View>
        <View
          style={
            Object {
              "height": 50,
              "width": 150,
            }
          }
        >
          <RNCPicker
            items={
              Array [
                Object {
                  "label": "Endzone",
                  "textColor": undefined,
                  "value": "zone",
                },
                Object {
                  "label": "3/4 field",
                  "textColor": undefined,
                  "value": "3/4 field",
                },
                Object {
                  "label": "Rectangle",
                  "textColor": undefined,
                  "value": "rectangle",
                },
                Object {
                  "label": "Empty",
                  "textColor": undefined,
                  "value": "empty",
                },
              ]
            }
            onChange={[Function]}
            selectedIndex={0}
            style={
              Array [
                Object {
                  "height": 216,
                },
                undefined,
              ]
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
